x-pack/plugins/cases/public/components/create/form_context.tsx

```diff
@@ -118,6 +152,7 @@ export const FormContext: React.FC<Props> = ({
             connector: connectorToUpdate,
             settings: { syncAlerts },
             owner: selectedOwner ?? defaultOwner,
+            customFields: transformedCustomFields as CaseUI['customFields'],
```

Comment:
We shouldn't need the type here. You can set the return type of `transformCustomFieldsData` to `CaseUI['customFields']` and fix any TS error.